doc/user/content/serve-results/s3.md

```diff
@@ -0,0 +1,231 @@
+               },
+               "Action": "sts:AssumeRole",
+               "Condition": {
+                   "StringEquals": {
+                       "sts:ExternalId": "mz_00000000-0000-0000-0000-000000000000_u0"
+                   }
+               }
+           }
+       ]
+   }
+   ```
+
+   Materialize **always uses the provided IAM principal** to assume the role, and
+   also generates an **external ID** which uniquely identifies your AWS connection
+   across all Materialize regions (see [AWS connection permissions](/sql/create-connection/#aws-permissions)).
+   For now, you'll set this ID to a dummy value; later, you'll update it with
+   the unique identifier for your Materialize region.
+
+1. Click **Next**.
+
+1. In **Add permissions**, select the IAM policy you created in [Create an IAM policy](#create-an-iam-policy),
+   and click **Next**.
+
+1. Enter a name for the role (e.g. `MZS3Exporter`), and click **Create role**.
+
+1. Click **View role** to see the role summary page, and note down the
+   role **ARN**. You will need it in the next step to create an AWS connection in
+   Materialize.
+
+## Step 2. Create a connection
+
+1. In the [SQL Shell](https://console.materialize.com/), or your preferred SQL
+   client connected to Materialize, create an [AWS connection](/sql/create-connection/#aws):
+
+   ```sql
+   CREATE CONNECTION aws_connection
+      TO AWS (ASSUME ROLE ARN = 'arn:aws:iam::400121260767:role/MZS3Exporter');
+   ```
+
+1. Retrieve the external ID for the connection:
+
+   ```sql
+   SELECT awsc.id, external_id
+    FROM mz_internal.mz_aws_connections awsc
+    JOIN mz_connections c ON awsc.id = c.id
+    WHERE c.name = 'aws_connection';
+   ```
+
+1. In your AWS account, find the IAM role you created in [Create an IAM role](#create-an-iam-role)
+   and, under **Trust relationships**, click **Edit trust policy**. Use the
+   `external_id` from the previous step to update the trust policy's
+   `ExternalId`, then click **Update policy**.
+
+   {{< warning >}}Failing to constrain the external ID in your role trust policy
+   will allow other Materialize customers to assume your role and use AWS
+   privileges you have granted the role!{{< /warning >}}
+
+1. Back in Materialize, validate the AWS connection you created using the
+   [`VALIDATE CONNECTION`](/sql/validate-connection) command.
+
+   ```sql
+   VALIDATE CONNECTION aws_connection;
+   ```
+
+   If no validation error is returned, you're ready to use this connection to
+   run a bulk export from Materialize to your target S3 bucket! 🔥
+
+## Step 4. Run a bulk export
+
+To export data to your target S3 bucket, use the [`COPY TO`](/sql/copy-to/#copy-to-s3)
+command, and the AWS connection you created in the previous step.
+
+{{< tabs >}}
+{{< tab "Parquet">}}
+
+```sql
+COPY some_object TO 's3://<bucket>/<path>'
+WITH (
+    AWS CONNECTION = aws_connection,
+    FORMAT = 'parquet'
+  );
```

Comment:
@rjobanp: if the command fails because of authentication issue, the error message is pretty cryptic.

`ERROR:  dispatch failure:  `

It's then possible to see the whole error message via `VALIDATE CONNECTION`:

```
 ERROR:  dispatch failure: other: an error occurred while loading credentials: service error: unhandled error (AccessDenied): Error { code: "AccessDenied", message: "User: arn:aws:sts::725780499770:assumed-role/MaterializeConnection/aws-eu-west-1-xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-x-xxxxx is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::001234567890:role/Materializes3Exporter", aws_request_id: "aef877b2-531e-486b-bd12-ab93090a6ddd" }
 ```
 
but it'd be nice to expose some more details in the actual error message.

Reply:
Damn, you're fast: #27362.